src/controllers/basic-messages/BasicMessageController.ts

```diff
@@ -48,4 +66,25 @@ export class BasicMessageController extends Controller {
       throw error
     }
   }
+
+  async waitForProof(connectionId: string, proofId: string, timeoutMs: number): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const timeout = setTimeout(() => {
+        reject(new HttpResponse({ message: `Proof request to connection '${connectionId}' timed out` }))
```

Comment:
It's good practice to remove the `onProofStateChanged` listener here as well. Although a promise can only resolve once, the listener will remain attached.

Reply:
added the removal